packages/amplify_core/lib/amplify_core.dart

```diff
@@ -80,5 +80,7 @@ class Amplify {
     if (!res) {
       throw ("Amplify plugin was not added");
     }
+
+    await DataStore.configure(configuration);
```

Comment:
Why is this being done outside of the `if (dataStorePlugins != null && dataStorePlugins.length == 1)` block?

Reply:
This is to configure the category, which may or may not decide to configure it's plugins. Think of it as javascript's `API.configure()` or `DataStore.configure()` to configure these categories outside of `Amplify.configure()` and we are using this opportunity here to configure the event channels for datastore category